### PR TITLE
feat(mgmt): added mgmt extension which returns current zot configuration

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andaaron — COMMENTED** [review]
This would work, I calling this API "mgmt" doesn't seem right to me. I don't see how this implementation fits into a holistic "mgmt" approach where in the future:
- users see part of the config based on their permissions
- users would also be able to POST a config based only on the fields they have permissions to set.

"mgmt" implies that you manage zot, meaning you would also be able to configure it in the future, while this implementation which skips all authentication on purpose (even for good reason) falls short with regards to a management API. 
If you don't have a way to specify which configuration you want based on query parameters/url, and you also can't figure it out based on the user permission, this "mgmt" API seems to be a dead end.

Sure. we can call it something else, which means expectations with regards to how this will evolve in the future would be different.

**Comment by peusebiu** [comment]
> This would work, I calling this API "mgmt" doesn't seem right to me. I don't see how this implementation fits into a holistic "mgmt" approach where in the future:
> 
> * users see part of the config based on their permissions
> * users would also be able to POST a config based only on the fields they have permissions to set.
> 
> "mgmt" implies that you manage zot, meaning you would also be able to configure it in the future, while this implementation which skips all authentication on purpose (even for good reason) falls short with regards to a management API. If you don't have a way to specify which configuration you want based on query parameters/url, and you also can't figure it out based on the user permission, this "mgmt" API seems to be a dead end.
> 
> Sure. we can call it something else, which means expectations with regards to how this will evolve in the future would be different.

auth is skipped only if you don't provide credentials, if you give it credentials then authn and authz will take care of the request.

**Comment by peusebiu** [comment]
modified the API response as follows:

no auth:
```
{
  "distSpecVersion": "1.1.0-dev",
  "binaryType": "",
  "http": {
    "auth": {}
  }
}
```
htpasswd auth:
```
{
  "distSpecVersion": "1.1.0-dev",
  "binaryType": "",
  "http": {
    "auth": {
      "htpasswd": {}
    }
  }
}
```

httpasswd + ldap:
```
{
  "distSpecVersion": "1.1.0-dev",
  "binaryType": "",
  "http": {
    "auth": {
      "htpasswd": {},
      "ldap": {
        "address": "ldap.example.org"
      }
    }
  }
}
```

ldap auth:
```
{
  "distSpecVersion": "1.1.0-dev",
  "binaryType": "",
  "http": {
    "auth": {
      "ldap": {
        "address": "ldap.example.org"
      }
    }
  }
}
```

bearer auth:
```
{
  "distSpecVersion": "1.1.0-dev",
  "binaryType": "-sync-search-scrub-metrics-lint-ui-mgmt",
  "http": {
    "auth": {
      "bearer": {
        "Realm": "https://auth.myreg.io/auth/token",
        "Service": "myauth"
      }
    }
  }
}
```

ldap + htpasswd + bearer auth:
```
{
  "distSpecVersion": "1.1.0-dev",
  "binaryType": "-sync-search-scrub-metrics-lint-ui-mgmt",
  "http": {
    "auth": {
      "htpasswd": {},
      "bearer": {
        "Realm": "https://auth.myreg.io/auth/token",
        "Service": "myauth"
      },
      "ldap": {
        "address": "ldap.example.org"
      }
    }
  }
}
```
@raulkele what do you think about this?

**Comment by raulkele** [comment]
The API response looks good to me

**Review by andaaron — DISMISSED** [review]

**Comment by andaaron** [comment]
> > This would work, I calling this API "mgmt" doesn't seem right to me. I don't see how this implementation fits into a holistic "mgmt" approach where in the future:
> > 
> > * users see part of the config based on their permissions
> > * users would also be able to POST a config based only on the fields they have permissions to set.
> > 
> > "mgmt" implies that you manage zot, meaning you would also be able to configure it in the future, while this implementation which skips all authentication on purpose (even for good reason) falls short with regards to a management API. If you don't have a way to specify which configuration you want based on query parameters/url, and you also can't figure it out based on the user permission, this "mgmt" API seems to be a dead end.
> > Sure. we can call it something else, which means expectations with regards to how this will evolve in the future would be different.
> 
> auth is skipped only if you don't provide credentials, if you give it credentials then authn and authz will take care of the request.

Thanks, it's clear now.

File: pkg/extensions/mgmt.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "ReleaseTag": "v1.1.15-rc1",
+  "BinaryType": "-sync-search-scrub-metrics-lint-mgmt",
+  "HTTP": {
+    "Auth": {
+      "FailDelay": 0,
+      "HTPasswd": {
+        "Path": "test/data/htpasswd"
+      },
+      "LDAP": {
+        "Address": "ldap.example.org"
```
Comment (andaaron) [inline pkg/extensions/mgmt.md:39]: This behavior was changed in the last amends, right?

Comment (peusebiu) [reply pkg/extensions/mgmt.md:39]: oh, yes, will fix

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]

**Pull request merged**